test/test_model.py

```diff
@@ -830,6 +830,9 @@ class NoNameStatus(ops.StatusBase):  # pyright: ignore[reportUnusedClass]
             class NonStringNameStatus(ops.StatusBase):  # pyright: ignore[reportUnusedClass]
                 name = None  # pyright: ignore[reportAssignmentType]
 
+        with pytest.deprecated_call():
```

Comment:
It feels like this would more properly be in a new (very small) test of its own, rather than being in the "base status instance" test.

```suggestion
    def test_statusbase_register_is_deprecated(self):
        with pytest.deprecated_call():
```